consistency-tests/__tests__/clickables.test.js

```diff
@@ -66,16 +63,80 @@ describe.each`
 
     it("sets the 'target' prop on the underlying element", () => {
         const wrapper = mount(
-            <Component
-                href="https://www.khanacademy.org"
-                target="_blank"
-                {...extraProps}
-            >
+            <Component href="https://www.khanacademy.org" target="_blank">
                 Click me
             </Component>,
         );
         wrapper.simulate("click");
 
         expect(wrapper.find("a")).toHaveProp("target", "_blank");
     });
+
+    it("renders a react-router Link if the URL is not external", () => {
+        const wrapper = mount(
+            <MemoryRouter>
+                <Component href="/foo/bar">Click me</Component>
+            </MemoryRouter>,
+        );
+        wrapper.simulate("click");
+
+        expect(wrapper.find(ReactRouterLink)).toExist();
+    });
+
+    it("does not render a react-router Link if the URL is external", () => {
+        const wrapper = mount(
+            <MemoryRouter>
+                <Component href="https://www.khanacademy.org/foo/bar">
+                    Click me
+                </Component>
+            </MemoryRouter>,
+        );
+        wrapper.simulate("click");
+
+        expect(wrapper.find(ReactRouterLink)).not.toExist();
+    });
+});
+
+// NOTE: Link doesn't work without an href so it isn't included in this suite
+describe.each`
+    Component            | name
+    ${ActionItem}        | ${"ActionItem"}
+    ${Button}            | ${"Button"}
+    ${ClickableWrapper}  | ${"Clickable"}
+    ${IconButtonWrapper} | ${"IconButton"}
+`("$name without an href", ({Component, name}) => {
```

Comment:
I added these tests since relying on snapshots to verify that the right underlying component is not ideal since it can take some time to validate the snapshots are correct.

Reply:
Nice set of tests!